src/parametric.jl

```diff
@@ -0,0 +1,64 @@
+parametrics = [
+    Arcsine,
+    Bernoulli,
+    Beta,
+    BetaBinomial,
+    BetaPrime,
+    Binomial,
+    Biweight,
+    Categorical,
+    Cauchy,
+    Chi,
+    Chisq,
+    Cosine,
+    Dirichlet,
+    Erlang,
+    Epanechnikov,
+    Exponential,
+    FDist,
+    FisherNoncentralHypergeometric,
+    Frechet,
+    Gamma,
+    GeneralizedPareto,
+    GeneralizedExtremeValue,
+    Geometric,
+    Gumbel,
+    InverseWishart,
+    InverseGamma,
+    InverseGaussian,
+    Laplace,
+    Levy,
+    Logistic,
+    LogNormal,
+    MvNormal,
+    MvNormalCanon,
+    MvTDist,
+    NegativeBinomial,
+    NoncentralBeta,
+    NoncentralChisq,
+    NoncentralF,
+    NoncentralHypergeometric,
+    NoncentralT,
+    Normal,
+    NormalCanon,
+    NormalInverseGaussian,
+    Pareto,
+    Poisson,
+    PoissonBinomial,
+    Rayleigh,
+    Skellam,
+    SymTriangularDist,
+    TDist,
+    TriangularDist,
+    Triweight,
+    Uniform,
+    VonMises,
+    WalleniusNoncentralHypergeometric,
+    Weibull,
+    Wishart
+]
+
+for D in parametrics
+    ex = :(@inline partype{T<:Real}(d::$(D.name.primary){T}) = T)
```

Comment:
I think it would be better to just have a one liner, e.g. `partype{T}(TDist{T})=T`,  for each of the parametric distributions in the place where they are defined. Usually, we shouldn't hook into the type internals like you do here.